codechain/primitives/PlatformAddress.py

```diff
@@ -0,0 +1,114 @@
+import re
+
+from ..crypto import bech32_decode
+from ..crypto import bech32_encode
+from ..crypto import blake160
+from .HexString import H160
+from .HexString import H512
+
+
+class PlatformAddress:
+    def __init__(self, account_id: H160, address: str):
+        self.account_id = account_id
+        self.value = address
+
+    @classmethod
+    def from_public(cls, public_key, **kwargs):
+        if not H512.check(public_key):
+            raise ValueError(
+                f"Invalid public key for creating PlatformAddress: {public_key}"
+            )
+
+        return cls.from_account_id(
+            get_account_id_from_public(str(public_key)), **kwargs
+        )
+
+    @classmethod
+    def from_account_id(cls, account_id, **kwargs):
+        if "network_id" in kwargs:
+            network_id = kwargs.get("network_id")
+        else:
+            network_id = "cc"
+        if "version" in kwargs:
+            version = kwargs.get("version")
+        else:
+            version = 1
+
+        if not H160.check(account_id):
```

Comment:
`H160.check` returns `True` if the format is correct regardless of type. It can be `str` or `H160`, but it contradicts with `__init__` requires `account_id` to be `H160`. You might want to use `account_id = H160(account_id)` to force the type, or just accept `H160` type.

Reply:
I changed function signature.
```
    def from_account_id(cls, account_id: H160, **kwargs):
```